### ISSUE TITLE
Add `curl` to redis docker image

### ISSUE THREAD

**Issue by guoard** [redis/docker-library-redis#406]
Redis Sentinel mode includes a configuration option that allows notifications (to an administrator or client) when a failover occurs. This feature is documented in the [sentinel.conf](https://github.com/redis/redis/blob/unstable/sentinel.conf#L254-L262).

When attempting to utilize this notification feature, I encountered a problem: `curl` is not installed in the image. I also attempted to use `wget` to send an HTTP request, but the `wget` included with busybox cannot handle HTTPS proxies.

Could `curl` be added to the redis docker image? This would enable the notification feature to function correctly by allowing HTTP requests to be sent reliably.

Thank you!

**Comment by guoard** [comment]
@adamiBs Yes, I did try using the official image as a base and adding `curl`, but maintaining a custom image adds overhead. We’d need to rebuild and test it with every Redis release, ensure multi-arch compatibility, and manage the CI/CD pipeline for it.

Including `curl` in the official image would make Sentinel's notification feature usable out of the box, with minimal added weight. It’s a small change that would simplify things for many users.

Thanks for considering it!

**Comment by Peter-Sh** [comment]
Hi @guoard!

Thanks for the suggestion and for explaining the Sentinel notification use case.

Unfortunately, we do not want to add curl or a full wget implementation to the Redis image by default. Even though this sounds like a small convenience, in practice it is not "just adding curl": it pulls in additional packages and shared libraries that are not part of Redis itself. For the Redis image, that adds roughly 10% to the image size, and more importantly it increases the amount of non-Redis software shipped in the image.

That extra software also becomes part of the image’s security and maintenance surface. Libraries needed by curl/wget can show up in vulnerability scans, require tracking and patching, and create noise or even deployment blockers for users scanning the image, despite those components not being related to the Redis product.

Because of that, we prefer to keep the official image focused on Redis and avoid adding general-purpose HTTP clients by default.
